src/groupByKey.ts

```diff
@@ -2,6 +2,33 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+type Value = string | number;
+
+type Obj = {
+  [key: string]: Value;
+};
+
+export function groupByKey(items: Obj[], key: keyof Obj): GroupsMap<Obj> {
+  const result: GroupsMap<Obj> = {};
+  const moddedItems = items.map((item1) => {
+    const arrayForItems: Obj[] = [];
+
+    items.filter((item2) => {
+      if (item2[key] === item1[key]) {
+        arrayForItems.push(item2);
+
+        return true;
+      }
+
+      return false;
+    });
+
+    return arrayForItems;
+  });
+
+  items.forEach((item, index) => {
+    result[item[key]] = moddedItems[index];
+  });
+
+  return result;
 }
```

Comment:
The logic of the function `groupByKey` is incorrect. The task requires to group items by key, but your function creates an array of arrays with all items from the original array. The resulting object has the same keys and arrays for each key. This is not what the task requires. Also, the code is inefficient because it uses unnecessary loops and operations. You can solve the task using one loop and without creating additional arrays. Here is an algorithm for solving the task: 1. Create an empty object `result`. 2. For each `item` in the `items` array do the following: 2.1. If `result` does not have a property with the name `item[key]`, then add this property to `result` and set its value to an empty array. 2.2. Push `item` to the array `result[item[key]]`. 3. Return `result`.